urlb57eb64056_jacadenac_ISI_Othello_UN_tgz-pJ8-othello_OthelloJ8/src/logica/Torneo.java

```diff
@@ -0,0 +1,131 @@
+package logica;
+
+import java.io.BufferedReader;
+import java.io.File;
+import java.io.FileNotFoundException;
+import java.io.FileReader;
+import java.io.FileWriter;
+import java.io.IOException;
+import java.io.PrintWriter;
+
+import javax.swing.JFileChooser;
+
+import core.Tablero.TipoTablero;
+
+public class Torneo {
+
+	private int cantidadVictorias;
+	public static int numeroTorneosCreados;
+	public String datosTorneo;
+	private static final File directorio = new File("setup/tournament");
+	
+	public Torneo(int[] estrategia, int[][] estrategiasCompetidoras, int tamTablero) {
+		super();
+		leerParametros();
+		datosTorneo="";
+		numeroTorneosCreados++;
+		cantidadVictorias=0;
+		iniciarTorneo(estrategia, estrategiasCompetidoras, tamTablero);
+	}
+
+	private void iniciarTorneo(int[] estrategia1, int[][] estrategiasCompetidoras, int tamTablero) {
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

The cyclomatic complexity of this method is 13. By comparison, 98% of the methods in the CodeGuru reference dataset have a lower cyclomatic complexity. This indicates the method has a high number of decisions and it can make the logic difficult to understand and test. We recommend that you simplify this method or break it into multiple methods.